soundata/annotations.py

```diff
@@ -36,39 +64,130 @@ class Events(Annotation):
             (as floats) in seconds in the form [start_time, end_time]
             with positive time stamps and end_time >= start_time.
         labels (list): list of event labels (as strings)
-        confidence (np.ndarray or None): array of float confidence values
-            in the range [0, 1]
+        confidence (np.ndarray or None): array of confidence values
+        labels_unit (str): labels unit, one of LABELS_UNITS
+        intervals_unit (str): intervals unit, one of TIME_UNITS
+
+
     """
 
-    def __init__(self, intervals, labels, confidence=None) -> None:
+    def __init__(
+        self,
+        intervals,
+        intervals_unit,
+        labels,
+        labels_unit,
+        confidence=None,
+    ) -> None:
+
         validate_array_like(intervals, np.ndarray, float)
         validate_array_like(labels, list, str)
         validate_array_like(confidence, np.ndarray, float, none_allowed=True)
         validate_lengths_equal([intervals, labels, confidence])
         validate_intervals(intervals)
         validate_confidence(confidence)
+        validate_unit(labels_unit, LABEL_UNITS)
+        validate_unit(intervals_unit, TIME_UNITS)
 
         self.intervals = intervals
+        self.intervals_unit = intervals_unit
         self.labels = labels
+        self.labels_unit = labels_unit
         self.confidence = confidence
 
 
+class SpatialEvents(Events):
+    """SpatialEvents class
+
+    Attributes:
+        intervals (np.ndarray): (n x 2) array of intervals
+            (as floats) in seconds in the form [start_time, end_time]
+            with positive time stamps and end_time >= start_time.
+        labels (list): list of event labels (as strings)
+        confidence (np.ndarray or None): array of confidence values
+        labels_unit (str): labels unit, one of LABELS_UNITS
+        intervals_unit (str): intervals unit, one of TIME_UNITS
+        azimuth (np.ndarray or None):
+        azimuth_unit (str): azimuth unit, one of AZIMUTH_UNITS
+        elevation (np.ndarray or None):
+        elevation_unit (str): elevation unit, one of AZIMUTH_UNITS
+        distance (np.ndarray or None):
+        distance_unit (str): distance unit, one of DISTANCE_UNITS
+        cartesian_coord (np.ndarray or None):
+        cartesian_coord_unit (str): cartesian_coord unit, one of DISTANCE_UNITS
```

Comment:
IIUC these azimuth, elevation, distance, and cartesian_coord are all arrays because these values can change over time.

This raises several complexities:
* What is the hop size in these time series? 
* Where is the hop size represented? 
* Presumably the series should correspond to the duration of the event's interval. How do we check for this? Will we enforce it?

@magdalenafuentes @iranroman

Reply:
You raise a good point here. If we interpret these annotations in the same way we interpret the `label` in an event, we would not know the underlying hop-size, but we would know beginning and end of the source being in position pi/2, pi/4 and so on. It doesn't sound right. If we want instead to move this to a hop-size-basis (e.g. like F0 in `mirdata`) and have times at a fixed hop-size values (which seems to make sense), then the connection with `Events` becomes a bit trickier? It seems confusing to have both `intervals` and `times` on it right? Unless we make `azimuth` ... named tuples which have time and value? Just brainstorming here

Reply:
Datasets with spatial events will often use a fixed time-step for all annotations. Hence, azimuth and elevation can be a vector of position values where elements are separated in time by the fixed time-step. Azimuth and elevation can also be a single value, which would indicate an event with a static position in space. 

I wrote these checks like this in my version of `SpatialEvents`:

```
def validate_positions(positions, units):
    """Validate if positions are well-formed.

    If positions is None, validation passes automatically

    Args:
        positions (np.ndarray): (n x 2) array
        units (string): 'degrees' or 'radians'

    Raises:
        ValueError: if positions have an invalid shape, the first column (azimuth) has
        a range of values exceeding 360 degrees, or if the second column (elevation) has
        a range of values exceeding 180 degrees

    """
    if positions is None:
        return

    # validate that positions have the correct shape
    positions_shape = np.shape(positions)
    if len(positions_shape) != 2 or positions_shape[1] != 2:
        raise ValueError(
            f"positions should be arrays with two columns, but array has {positions_shape}"
        )

    # validate that the first column has a range of values less than 360 degrees
    if (np.ptp(positions[:,0]) > 360) and units == 'degrees':
        raise ValueError(f"Range of azimuth values should be less than 360 degrees")

    # validate that the second column has a range of values less than 180 degrees
    if (np.ptp(positions[:,1]) > 180) and units == 'degrees':
        raise ValueError(f"Range of elevation values should be less than 180 degrees")
```